pymongo/mongo_client.py

```diff
@@ -2393,12 +2384,12 @@ def _write(self) -> T:
         try:
             max_wire_version = 0
             self._server = self._get_server()
-            supports_session = (
-                self._session is not None and self._server.description.retryable_writes_supported
-            )
             with self._client._checkout(self._server, self._session) as conn:
                 max_wire_version = conn.max_wire_version
-                if self._retryable and not supports_session:
+                sessions_supported = (
+                    self._server.description.retryable_writes_supported and conn.supports_sessions
```

Comment:
Is `self._session` never `None` anymore?

Reply:
Added it back for safety.